dstep/translator/MacroDefinition.d

```diff
@@ -0,0 +1,1429 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Jun 03, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+module dstep.translator.MacroDefinition;
+
+import std.array : Appender;
+
+import clang.c.Index;
+import clang.Cursor;
+import clang.Token;
+
+import dstep.translator.Context;
+import dstep.translator.Output;
+
+/**
+  * The accept family of functions parse tokens. In the case of successful parsing
+  * advances beginning of tokens to the next token and returns true.
+  *
+  * The parsing is successful, if the first token in tokens has a kind kind and
+  * its spelling matches on of the strings passed as Args.
+  * It assigns the spelling of the token to the spelling parameter.
+  */
+
+bool accept(Args...)(ref TokenRange tokens, ref string spelling, TokenKind kind) if (Args.length > 0)
```

Comment:
Most of my concern is that the API of the function is not very clear. There's no indication in the documentation or in a template constraint that `Args` should only be strings. The name `Args` is also very generic, i.e. that you could pass in any type of arguments.